pkg/kube/client.go

```diff
@@ -542,9 +543,12 @@ func perform(infos ResourceList, fn func(*resource.Info) error) error {
 	for range infos {
 		err := <-errs
 		if err != nil {
-			return err
+			errList = append(errList, err)
 		}
 	}
+	if errList != nil {
+		return errList[0]
```

Comment:
Why would you save all the errors and just use the first?

Reply:
HI, Howe,  I don't find a proper way to keep the return type not be changed and  return all the errors. I think the import thing is avoid gorotine leak here. If you have other ideas, please tell me,thanks.

Reply:
`var lastError error` and just return it instead of `return nil`

Reply:
yes, use lastError looks good, thanks.